File: Recursion/python/1991.py
n = int(input())
tree = {}
for _ in range(n):
    root_node,left,right = input().split()
    tree[root_node] = [left,right];

def preorder (node):
    if (node != "."):
        print(node,end="");
        preorder(tree[node][0])
        preorder(tree[node][1])
def inorder (node):
    if (node != "."):
        inorder(tree[node][0])
        print(node,end="");
        inorder(tree[node][1])
def postorder (node):
    if (node != "."):
        postorder(tree[node][0])
        postorder(tree[node][1])
        print(node,end="");

preorder("A")
print()
inorder("A")
print()
postorder("A")
print()

File: Recursion/javascript/1991.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);
let tree = new Map();
let rootNode, leftNode, rightNode;
for (let i = 1 ; i<input.length-1 ; i++){
  [rootNode, leftNode, rightNode] = input[i].split(' ');
  tree.set(rootNode,[leftNode,rightNode]);
}

const preorderTraversal = (node) =>{
  if (node ===".") return
  answer += node;
  preorderTraversal(tree.get(node)[0]);
  preorderTraversal(tree.get(node)[1]);
}

const inorderTraversal = (node) =>{
  if (node ===".") return
  inorderTraversal(tree.get(node)[0]);
  answer += node;
  inorderTraversal(tree.get(node)[1]);
}

const postorderTraversal = (node) =>{
  if (node ===".") return
  postorderTraversal(tree.get(node)[0]);
  postorderTraversal(tree.get(node)[1]);
  answer += node;
}

let answer = "";
preorderTraversal("A");
console.log(answer);

answer = "";
inorderTraversal("A");
console.log(answer);

answer = "";
postorderTraversal("A");
console.log(answer);